File: supcoder-dashboard/src/main/kotlin/com/supcoder/hub/dashboard/controller/UserController.java
package com.supcoder.hub.dashboard.controller;

import com.supcoder.hub.core.util.JsonResult;
import com.supcoder.hub.dashboard.auth.JwtUtil;
import com.supcoder.hub.db.domain.User;
import com.supcoder.hub.db.service.UserService;
import com.supcoder.hub.core.util.ResultUtil;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/user")
public class UserController {

    @Autowired
    private UserService userService;

    @Autowired
    private JwtUtil jwtUtil;

    @GetMapping("/currentUser")
    public ResponseEntity<?> getCurrentUser(@RequestHeader(value = "Authorization", required = false) String authorization) {
        if (authorization == null) {
            return ResponseEntity.badRequest().body("Authorization header is missing");
        }
        String username = jwtUtil.extractUsername(authorization);
        if (username == null) {
            return ResponseEntity.badRequest().body("Invalid authorization token");
        }
        User user = userService.queryByUsername(username);
        return ResponseEntity.ok(ResultUtil.success(user));
    }

    @GetMapping("/{id}")
    public ResponseEntity<JsonResult<User>> findUserById(@PathVariable Long id) {
        return ResponseEntity.ok(ResultUtil.success(userService.findById(id)));
    }

    @PostMapping
    public ResponseEntity<JsonResult<String>> saveUser(@RequestBody User user) {
        userService.add(user);
        return ResponseEntity.ok(ResultUtil.success("User saved successfully"));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<JsonResult<String>> deleteUserById(@PathVariable Long id) {
        userService.deleteById(id);
        return ResponseEntity.ok(ResultUtil.success("User deleted successfully"));
    }

    @DeleteMapping("username/{username}")
    public ResponseEntity<JsonResult<String>> deleteUserByUserName(@PathVariable String username) {
        userService.deleteByUserName(username);
        return ResponseEntity.ok(ResultUtil.success("User deleted successfully"));
    }

    @GetMapping("/username/{username}")
    public ResponseEntity<JsonResult<User>> findByUsername(@PathVariable String username) {
        return ResponseEntity.ok(ResultUtil.success(userService.queryByUsername(username)));
    }


    @GetMapping("/list")
    public ResponseEntity<JsonResult<List<User>>> getUserList(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "10") int size) {
        List<User> userList = userService.querySelective(null, page, size, null, null);
        return ResponseEntity.ok(ResultUtil.success(userList));
    }

}

File: supcoder-dashboard/src/main/kotlin/com/supcoder/hub/dashboard/controller/UserController.kt
package com.supcoder.hub.dashboard.controller


import com.supcoder.hub.core.util.JsonResult
import com.supcoder.hub.dashboard.model.vo.City
import com.supcoder.hub.dashboard.model.vo.CurrentUser
import com.supcoder.hub.dashboard.model.vo.Geographic
import com.supcoder.hub.dashboard.model.vo.Province
import com.supcoder.hub.dashboard.model.vo.Tag
import com.supcoder.hub.db.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import com.supcoder.hub.core.util.ResultUtil

@RestController
@RequestMapping("/api/users")
class UserController {

    @Autowired
    private lateinit var userService: UserService


    @GetMapping("/currentUser")
    fun getCurrentUser(): ResponseEntity<JsonResult<CurrentUser>> {
        // 实现获取当前用户的逻辑
        val currentUser = CurrentUser(
            name = "John Doe",
            avatar = "http://example.com/avatar.jpg",
            userid = "12345",
            email = "john.doe@example.com",
            signature = "Hello, world!",
            title = "Developer",
            group = "Engineering",
            tags = listOf(Tag("dev", "Developer")),
            notifyCount = 5,
            unreadCount = 3,
            country = "China",
            access = "admin",
            geographic = Geographic(
                province = Province(label = "Guangdong", key = "gd"),
                city = City(label = "Shenzhen", key = "sz")
            ),
            address = "123 Main St",
            phone = "1234567890"
        )
        return ResponseEntity.ok(ResultUtil.success(currentUser)) as ResponseEntity<JsonResult<CurrentUser>>
    }

//    @GetMapping
//    fun findAllUsers(): ResponseEntity<JsonResult<List<User>>> {
//        return ResponseEntity.ok(ResultUtil.success(userService.findAll()))
//    }

    @GetMapping("/{id}")
    fun findUserById(@PathVariable id: Long): ResponseEntity<JsonResult<User>> {
        return ResponseEntity.ok(ResultUtil.success(userService.findById(id))) as ResponseEntity<JsonResult<User>>
    }

    @PostMapping
    fun saveUser(@RequestBody user: User): ResponseEntity<JsonResult<String>> {
        userService.add(user)
        return ResponseEntity.ok(ResultUtil.success("User saved successfully")) as ResponseEntity<JsonResult<String>>
    }

    @DeleteMapping("/{id}")
    fun deleteUserById(@PathVariable id: Long): ResponseEntity<JsonResult<String>> {
        userService.deleteById(id)
        return ResponseEntity.ok(ResultUtil.success("User deleted successfully")) as ResponseEntity<JsonResult<String>>
    }

    @GetMapping("/by-username/{username}")
    fun findByUsername(@PathVariable username: String): ResponseEntity<JsonResult<User>> {
        return ResponseEntity.ok(ResultUtil.success(userService.queryByUsername(username))) as ResponseEntity<JsonResult<User>>
    }
}
